raiden/network/transport/matrix.py

```diff
@@ -280,6 +280,11 @@ def stop_and_wait(self):
             self._client.greenlets.clear()
             self._client.logout()
 
+    @property
+    def greenlet(self) -> gevent.Greenlet:
```

Comment:
I think this is a bad approach, why don't you make the `RaidenService` and `Transport` classes  greenlets, and let them manage their child tasks instead of exposing it?

Reply:
@hackaugusto I thought about it too, but these classes aren't greenlets on theirselves, as there's no obvious loop inside (`MatrixTransport` depends on the underlying `_client.sync_thread` greenlet, `RaidenService` doesn't actually have a loop/long-lasting `_run` method or alike, `APIServer` depends on the `WSGIServer` instance, which isn't even a greenlet on itself nor have one, but use low-level gevent calls to asynchronously dispatch request handling, that greenlet inside it just wraps the busy wait).
Rather, they _control_ one or more subtasks/greenlets, like `AlarmTask`, `GMatrixClient` and `WSGIServer`, and are the responsible for starting/stopping them and being a point of control if the underlying task fails unexpectedly. How about a `RaidenTask` abstract class, with `start`, `stop`, possibly `join` methods and a `long_greenlets() -> List[Greenlet]` property?

Reply:
An option would also to make a small `StoppableGreenlet` abstract class (greenlet with synchronous `stop` method), and make those classes inherint from it, making them effectivelly greenlets (as you asked), but in those cases, the `_run` method would be a simple `get` on the children tasks, which would make it re-raise underlying exceptions, if any, and cleanup and exit cleanly when `stop` is called

Reply:
Yeah, having a standard method to stop all these classes is good, I just want to avoid the introspection used here:

https://github.com/raiden-network/raiden/blob/d8839aee71d85e2d9c0617626001c36b17f82fc4/raiden/ui/cli.py#L1015-L1025